src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs

```diff
@@ -47,14 +55,18 @@ public static IServiceCollection AddApplicationInsights(this IServiceCollection
 
             services.AddSingleton<ITelemetryInitializer, HttpDependenciesParsingTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, WebJobsRoleEnvironmentTelemetryInitializer>();
-            services.AddSingleton<ITelemetryInitializer, WebJobsTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, WebJobsSanitizingInitializer>();
+            services.AddSingleton<ITelemetryInitializer, WebJobsTelemetryInitializer>();
+            services.AddSingleton<ITelemetryInitializer>(new SdkVersionRequestTelemetryInitializer(sdkVersion));
```

Comment:
Can you make this a factory so DI will "own" it if it ever needs disposing? 

            services.AddSingleton<ITelemetryInitializer>(_ => new SdkVersionRequestTelemetryInitializer(sdkVersion));